pytorch_lightning/callbacks/early_stopping.py

```diff
@@ -229,7 +229,7 @@ def _stop_distributed_training(self, trainer, pl_module):
 
         if trainer.use_tpu:
             stop = torch.tensor(int(trainer.should_stop), device=pl_module.device, dtype=torch.int32)
-            stop = xm.mesh_reduce("stop_signal", stop, torch.cat)
+            stop = xm.mesh_reduce("stop_signal", stop, sum)
```

Comment:
does this solve https://github.com/PyTorchLightning/pytorch-lightning/issues/2903 issue??

Reply:
I don't think so. This issue was related to the `stop` tensor being a 0-D tensor. Running `torch.cat` on a list of 0-D resulted in an error `RuntimeError: zero-dimensional tensor (at position 0) cannot be concatenated`. I think `sum` was intended here instead.

Reply:
oh, wait. It does fix this issue on the notebook mentioned there.

<img width="1135" alt="Screenshot 2020-08-30 at 9 41 59 AM" src="https://user-images.githubusercontent.com/6305654/91651050-22a58e00-eaa5-11ea-8c19-3cb048385226.png">

Reply:
There is a new error coming now in the notebook, but I guess that is a different issue.

Reply:
wondering how is it possible that the error comes in Notebook but here it passes

Reply:
not sure, but I have observed tpu training usually fails either during the last epoch(>1) or 2nd epoch. but in tests `max_epochs` is set to 1.